Kpng local ipvs

jayunit100 opened the pull request:
update local instructions to support pull policy, loading kind images locally rather then dockerhub

jayunit100 commented:
bump @mcluseau  :) cide from fridays mtng  for folks to test ipvs out

mcluseau commented on hack/kpng-local-up.sh:
```diff
@@ -1,8 +1,10 @@
 
-: ${IMAGE:="jayunit100/kpng-server:latest"}
-
+: ${IMAGE:="jayunit100/kpng:ipvs"}
```
plz add 2 more defaults:
```
: ${PULL:=Always}
: ${BACKEND:=nft}
```

mcluseau commented on hack/kpng-local-up.sh:
```diff
@@ -1,8 +1,10 @@
-: ${IMAGE:="jayunit100/kpng-server:latest"}
-
+: ${IMAGE:="jayunit100/kpng:ipvs"}
+echo $IMAGE
```
change to -n `echo this will deploy kpng with docker image $IMAGE, pull policy $PULL and the $BACKEND backend. Press enter to confirm, C-c to cancel`

mcluseau commented on hack/kpng-local-up.sh:
```diff
@@ -1,8 +1,10 @@
-
+: ${IMAGE:="jayunit100/kpng:ipvs"}
+echo $IMAGE
+echo $PULL
```
remove

mcluseau replied:
the sed part must be updated accordingly

mcluseau commented on hack/kpng-deployment-ds.yaml.tmpl:
```diff
@@ -51,9 +53,7 @@ spec:
-        command:
-        - kpng-nftables
-        - --target=unix:///k8s/proxy.sock
+        args: [ "local", "--api=unix:///k8s/proxy.sock", "to-ipvs" ]
```
change `to-ipvs` to `to-KPNG_BACKEND`

mcluseau commented on hack/README.md:
````diff
@@ -17,6 +17,19 @@ However it cant use a vanilla kind recipe because:
+To run kpng from source, you can run
+```
+docker build -t myname/kpng:ipvs ./
+KPNG_IMAGE=myname/kpng:ipvs KPNG_PULL=Never ./kpng-local-up.sh
````
should be IMAGE=... PULL=... BACKEND=... (nft or ipvs; you choose ;))

jayunit100 commented:
fixed, testing nowwwwww

jayunit100 commented:
addressed comments... 
``` docker build -t myname/kpng:ipvs ./ ; IMAGE=myname/kpng:ipvs PULL=Never IMAGE=myname:kpng:ipvs BACKEND=ipvs ./kpng-local-up.sh ; kind load docker-image myname/kpng:ipvs --name=kpng-proxy ```
output
```
jayunit100-hyper@jayunit100-hyp:~/SOURCE/kpng$ kubectl get pods -A
NAMESPACE            NAME                                               READY   STATUS    RESTARTS   AGE
kube-system          calico-kube-controllers-6949477b58-h64l4           1/1     Running   0          16d
kube-system          calico-node-2q6fb                                  1/1     Running   0          16d
kube-system          calico-node-84xtz                                  1/1     Running   0          16d
kube-system          calico-node-pr2q7                                  1/1     Running   0          16d
kube-system          coredns-74ff55c5b-9td48                            1/1     Running   0          16d
kube-system          coredns-74ff55c5b-k45g9                            1/1     Running   0          16d
kube-system          etcd-kpng-proxy-control-plane                      1/1     Running   0          16d
kube-system          kpng-lq5bp                                         2/2     Running   0          16d
kube-system          kpng-mq78n                                         2/2     Running   0          16d
kube-system          kpng-pvr6d                                         2/2     Running   0          16d
kube-system          kube-apiserver-kpng-proxy-control-plane            1/1     Running   0          16d
kube-system          kube-controller-manager-kpng-proxy-control-plane   1/1     Running   0          16d
kube-system          kube-scheduler-kpng-proxy-control-plane            1/1     Running   0          16d
local-path-storage   local-path-provisioner-78776bfc44-f9xtq            1/1     Running   0          16d

```

jayunit100 commented on hack/kpng-local-up.sh:
```diff
@@ -39,7 +43,9 @@ function build {
     # substitute it with your changes...
     echo "Applying template with KPNG_IMAGE=$IMAGE"
-    cat kpng-deployment-ds.yaml.tmpl | sed "s,KPNG_IMAGE,$IMAGE," > kpng-deployment-ds.yaml
+    cat kpng-deployment-ds.yaml.tmpl | sed "s,IMAGE,$IMAGE," > kpng-deployment-ds.yaml.1
```
Just for simplicity, dont want to overabstract this yet.
someday we'll have ytt or kustomize or whatever here, or just have a go launcher

mcluseau commented:
those 16d pods are suspect ;) I'll merge, double check and do the changes I have in mind :)
